File: java/src/main/java/problems1_100/SortColors.java
package problems1_100;

enum SortColors {
    DEFAULT {
        @Override
        void sortColors(int[] nums) {
            for (int i = 0; i < nums.length - 1; i++) {
                for (int j = i + 1; j < nums.length; j++) {
                    if (nums[j] >= nums[i]) {
                        continue;
                    }
                    int temp = nums[j];
                    nums[j] = nums[i];
                    nums[i] = temp;
                }
            }
        }
    };

    abstract void sortColors(int[] nums);
}

File: kotlin/src/main/kotlin/problems1_100/SortColors.kt
package problems1_100

enum class SortColors {
    DEFAULT {
        override fun sortColors(nums: IntArray) {
            for (i in 0 until nums.size - 1) {
                for (j in i + 1 until nums.size) {
                    if (nums[j] >= nums[i]) {
                        continue
                    }
                    val temp = nums[j]
                    nums[j] = nums[i]
                    nums[i] = temp
                }
            }
        }
    },
    ;

    abstract fun sortColors(nums: IntArray)
}
